src/actions/channels.ts

```diff
@@ -724,6 +724,36 @@ export function deleteChannel(channelId: string): ActionFunc {
     };
 }
 
+export function unarchiveChannel(channelId: string): ActionFunc {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        let state = getState();
+        const viewArchivedChannels = state.entities.general.config.ExperimentalViewArchivedChannels === 'true';
```

Comment:
Can you move this check to happen after the client call? While it's unlikely for the config, the state could change during between now and after the client call, so we'd want to check the correct value at that time